annif/project.py

```diff
@@ -217,6 +218,14 @@ def dump(self):
                 'backend': {'backend_id': self.config['backend']}
                 }
 
+    def remove_model_data(self):
+        """remove the data of this project"""
+        datadir_path = self._datadir_path
+        if not os.path.isdir(datadir_path):
+            raise AnnifException('No model data to remove for project {}.'
```

Comment:
I think this should emit a warning message but it shouldn't be a fatal error.

Reply:
Implemented showing warning instead of erroring. 

In the test for nonexistent data the command is run for `dummy-fi` project, because when running for a nonexistent project a (different) warning is raised already in [`get_project`](https://github.com/NatLibFi/Annif/blob/master/annif/cli.py#L31). 

The test has `logger.propagate = True` for the same reason as in https://github.com/NatLibFi/Annif/pull/299#discussion_r313433563.